### ISSUE TITLE
"No such file or directory" when using the built-in filesystem cache

### ISSUE THREAD

**Issue by mattpolicastro** [cgiffard/node-simplecrawler#175]
The built-in caching tool appears to be broken; I've tried caching to both existing and non-existing directories and received the following errors (`fs.mkdirSync()` and other synchronous writes?):

````
// test.js
var Crawler = require('simplecrawler');

var cacheDir = 'google' + '/';
var url = "http://www.google.com";

var crawler = Crawler.crawl(url);

crawler.cache = new Crawler.cache(cacheDir);
crawler.on('fetchcomplete', function(queueItem) {
  console.log(queueItem.url);
});
````

````
$ node test.js http://www.google.com
fs.js:747
  return binding.mkdir(pathModule._makeLong(path),
                 ^
Error: ENOENT, no such file or directory 'google/http'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:747:18)
    at /Users/mpolicastro/crawler/node_modules/simplecrawler/lib/cache-backend-fs.js:203:16
    at Array.forEach (native)
    at FSBackend.setItem (/Users/mpolicastro/crawler/node_modules/simplecrawler/lib/cache-backend-fs.js:187:15)
    at EventEmitter.Cache.setCacheData (/Users/mpolicastro/crawler/node_modules/simplecrawler/lib/cache.js:30:20)
    at processReceivedData (/Users/mpolicastro/crawler/node_modules/simplecrawler/lib/crawler.js:1036:27)
    at IncomingMessage.receiveData (/Users/mpolicastro/crawler/node_modules/simplecrawler/lib/crawler.js:1106:13)
    at IncomingMessage.emit (events.js:129:20)
    at _stream_readable.js:908:16
````

````
$ mkdir -p google/http
$ node test.js http://www.google.com
test
/Users/mpolicastro/crawler/node_modules/simplecrawler/lib/cache-backend-fs.js:161
                throw error;
                      ^
Error: EISDIR, open 'google/http/www.google.com/80/index.html'
    at Error (native)
````

**Edit:** Forgot to include the sample script for reproducing the issue.
**Edit 2:** Had a small typo in the test script (had named the cache dir to `test`, not `google` as can be seen in the error logs).

**Comment by matthewforr** [comment]
I'm seeing a similar issue.

**Comment by cgiffard** [comment]
The cache code is more than a little crusty. I'll stick this on my todo list. Sorry all. :(

**Comment by matthewforr** [comment]
No problem, from what I can gather, the issue is that the code is trying to
write the file as a folder. I have it working by using an older version of
node-simplecrawler

**Comment by mattpolicastro** [comment]
@matthewforr which version, might I ask? I've tried rolling back a few versions to no avail, and would kill to have something to work with in the interim.

**Comment by cgiffard** [comment]
Well, it definitely used to work at some point, but I have a suspicion it's been broken for quite a while. There are no tests for it, so it kinda figures.

**Comment by cgiffard** [comment]
A desire to fix this issue dates all the way back to #4, haha. I really need to make it work properly.
